File: win-prog/ProgWinVB/Menus/ContextMenuAdd/ContextMenuAdd.vb
'-----------------------------------------------
' ContextMenuAdd.vb (c) 2002 by Charles Petzold
'-----------------------------------------------
Imports System
Imports System.Drawing
Imports System.Windows.Forms

Class ContextMenuAdd
    Inherits Form

    Private miColor As MenuItem

    Shared Sub Main()
        Application.Run(New ContextMenuAdd())
    End Sub

    Sub New()
        Text = "Context Menu Using Add"

        Dim cm As New ContextMenu()
        Dim eh As EventHandler = AddressOf MenuColorOnClick
        Dim mi As MenuItem

        cm.MenuItems.Add("Black", eh)
        cm.MenuItems.Add("Blue", eh)
        cm.MenuItems.Add("Green", eh)
        cm.MenuItems.Add("Cyan", eh)
        cm.MenuItems.Add("Red", eh)
        cm.MenuItems.Add("Magenta", eh)
        cm.MenuItems.Add("Yellow", eh)
        cm.MenuItems.Add("White", eh)

        For Each mi In cm.MenuItems
            mi.RadioCheck = True
        Next mi

        miColor = cm.MenuItems(3)
        miColor.Checked = True
        BackColor = Color.FromName(miColor.Text)

        ContextMenu = cm
    End Sub

    Private Sub MenuColorOnClick(ByVal obj As Object, _
                                 ByVal ea As EventArgs)
        miColor.Checked = False
        miColor = DirectCast(obj, MenuItem)
        miColor.Checked = True
        BackColor = Color.FromName(miColor.Text)
    End Sub
End Class

File: win-prog/ProgWinCS/Menus/ContextMenuAdd/ContextMenuAdd.cs
//---------------------------------------------
// ContextMenuAdd.cs © 2001 by Charles Petzold
//---------------------------------------------
using System;
using System.Drawing;
using System.Windows.Forms;

class ContextMenuAdd: Form
{
     MenuItem miColor;

     public static void Main()
     {
          Application.Run(new ContextMenuAdd());
     }
     public ContextMenuAdd()
     {
          Text = "Context Menu Using Add";

          ContextMenu  cm = new ContextMenu();
          EventHandler eh = new EventHandler(MenuColorOnClick);

          cm.MenuItems.Add("Black",   eh);
          cm.MenuItems.Add("Blue",    eh);
          cm.MenuItems.Add("Green",   eh);
          cm.MenuItems.Add("Cyan",    eh);
          cm.MenuItems.Add("Red",     eh);
          cm.MenuItems.Add("Magenta", eh);
          cm.MenuItems.Add("Yellow",  eh);
          cm.MenuItems.Add("White",   eh);

          foreach (MenuItem mi in cm.MenuItems)
               mi.RadioCheck = true;

          miColor = cm.MenuItems[3];
          miColor.Checked = true;
          BackColor = Color.FromName(miColor.Text);

          ContextMenu = cm;
     }
     void MenuColorOnClick(object obj, EventArgs ea)
     {
          miColor.Checked = false;
          miColor = (MenuItem) obj;
          miColor.Checked = true;

          BackColor = Color.FromName(miColor.Text);
     }
}
